### ISSUE TITLE
Use a fine-tuned pretrained model to classify new images

### ISSUE THREAD

**Issue by vic4ever** [facebook/fb.resnet.torch#130]
I have fine-tuned the ResNet-50 model with my own data which have 3 classes. I guess this will replace the last layer with a 3-way classifier.  I want to test the new model (the fine-tuned ResNet-50 model with the 3-way classifier) on my test set. From my understanding of the code, there are some possible ways to do it.

- Using classify.lua: As classify.lua takes a model such as ResNet-50 as input, I need to find a way to save the new model first then use classify.lua with the new model. My question here is how to save the fine-tuned ResNet-50 model with the 3-way classifier? 
- Using testOnly: this option makes train.lua to run only on the validation set, which I guess I can point to my test set. My concern is that which model will train.lua use. I guess it is the model set by the retrain option. Is it possible for me to fine-tune the ResNet-50 model and use the fine-tuned model with the testOnly option?

Thanks.

**Comment by colesbury** [comment]
Yeah the second way is what you want: use `-testOnly` combined with `-retrain`. You'll need to point it at the test set too.

**Comment by vic4ever** [comment]
Thanks for your reply. Following your advice, I have written my own DataLoader which can handle both training and testing data. 
```
-- Data loading
local trainLoader, valLoader, testLoader = DataLoader.create(opt)

if opt.testOnly then
   local top1Err, top5Err = trainer:test(0, testLoader )
   print(string.format(' * Results top1: %6.3f  top5: %6.3f', top1Err, top5Err))
   return
end
```
I have also added the following code in main.lua:64 so that the best retrained model is saved.
```
print("Saving model")
torch.save( paths.concat(opt.gen,"bestmodel.t7"),model)
```
The following command will retrain the ResNet-50 model on my data and also save the best retrained model in opt.gen.
`th main.lua -retrain resnet-50.t7 -data ~/data -resetClassifier true -nClasses 3 -dataset myData`
The ~/data folder contains 3 subfolders: train, val and test. 

Then, I can test the retrained model with my test set using this command: 
`th main.lua -retrain ./gen/bestmodel.t7 -data ~/data  -dataset myData -testOnly true`
Note that in this command, the `-retrain` model is not the ResNet model.

**Comment by colesbury** [comment]
That makes sense. The code should already save the best model in a file called `model_best.t7` (see [checkpoints.lua#L64-64](https://github.com/facebook/fb.resnet.torch/blob/fd4f1a7ac25141d3d9acd0ff3e6fac90800fc261/checkpoints.lua#L64-L66))

**Comment by vic4ever** [comment]
Thanks 👍
